File: palindrome/Palindrome.java
public class Palindrome {

  public static boolean palindrome(String word) {
    String reverse = new StringBuffer(word).reverse().toString();
    return word.equals(reverse);
  }

  public static void main(String[] args) {
    // Palindrome
    System.out.println(palindrome("bob"));
    // Not a Palindrome
    System.out.println(palindrome("qwerty"));
  }

}

File: palindrome/palindrome.cpp
#include <iostream>
#include <string>

using namespace std;

bool palindrome(string word)
{
  string rev = string(word.rbegin(), word.rend());
  return word.compare(rev) == 0 ? true : false;
}

int main()
{
  // Palindrome
  cout << palindrome("bob") << endl;

  // Not a Palindrome
  cout << palindrome("qwerty") << endl;
  return 0;
}
